src/axolotl/core/optimizers/adopt.py

```diff
@@ -0,0 +1,510 @@
+# mypy: allow-untyped-decorators
```

Comment:
Please add citation https://github.com/iShohei220/adopt/tree/main?tab=readme-ov-file#citation

Reply:
What is the proper way to add citation?